searchapp/views.py

```diff
@@ -402,12 +405,18 @@ def get_generic_paper(request):
 
 
 def get_customize_paper(request):
+    if not request.user.is_authenticated:
+        return JsonResponse({"error": "To access customised paper, you need to be logged in"}, status=401)
     if request.method == 'POST':
         subject = request.POST['subject']
         # filter returns a query set which is converted to a list and then the first element is picked up.
         subject_name = list(Subject.objects.filter(
             id=subject).values_list('subject_name', flat=True))[0]
-        chapters = map(int, request.POST.getlist('chapters[]')[0].split(','))
+        chapters = request.POST.getlist('chapters[]')[0].split(',')
+        if chapters[0]:
```

Comment:
This will throw an IndexError if chapters is empty.

Reply:
If `chapters[][0]` is empty, using `.split(',')` on it will return a list with one element, which is an empty string. So, this line basically checks that condition itself.

```
>>> x = ''.split(',')
>>> x[0]
''
```